Correct errors caught by -Werror=format-security
Patch author: Ryan Finnie <ryan@finnie.org>

-----

Flush logs when running non-interactive
Patch author: Alexander Inyukhin <shurick@sectorb.msk.ru>

See http://bugs.debian.org/691069

--- a/src/digitemp.c
+++ b/src/digitemp.c
@@ -586,7 +586,8 @@
       perror("Error loging to logfile");
     close( fd );
   } else {
-    printf( line );
+    printf( "%s",line );
+    fflush( stdout );
   }
   return 0;
 }  
--- a/userial/ioutil.c
+++ b/userial/ioutil.c
@@ -86,7 +86,7 @@
    // remember the start length
    deflen = strlen(buf);
    if (deflen < 80)
-      sprintf(defbuf,buf);
+      sprintf(defbuf,"%s",buf);
    else
       defbuf[0] = 0;
 
